query_temp.py

```diff
@@ -1,186 +1,113 @@
-            Default value is 0.0025 or the 2.5th percentile
-        month (int, optional): desired month in which to calculate
-            the desired percentile, default is 7, or July.
-    Returns:
-        query (sqlalchemy query): sqlalchemy query object
-            for design value
-    """
-
+    # percentile calculation
     percentile = (func.percentile_cont(quantile)
-                 .within_group(Obs.datum.desc())
-                 .label("air_temperature"))
-    days = days_in_month(start_time, end_time)
-    completeness = (count(Obs)/days).label("completeness")
-
-    month = start_time.month
-    # construct query table
-    query = (
-             session.query(percentile,
-                           min_time(Obs),
-                           max_time(Obs),
-                           History.lat,
-                           History.lon,
-                           History.station_id,
-                           completeness)
+                      .within_group(order)
+                      .label("air_temperature"))
+
+    # This query has eligible stations with fully complete
+    # record over month of interest
+    cte_full_cvg = (session.query(
+                            completeness,
+                            History.station_id.label('station_id'),
+                            func.extract('year', Obs.time).label('year')
+                            )
+                           .join(Variable, Obs.vars_id == Variable.id)
+                           .join(History, Obs.history_id == History.id)
+                           .filter(and_(Obs.time >= start_time,
+                                        Obs.time < end_time))
+                           .filter(func.extract('month', Obs.time) == start_time.month)
+                           .filter(Variable.name == net_var_name)
+                           .filter(and_(Variable.standard_name == standard_name,
+                                        Variable.cell_method == cell_method))
+                           .having(completeness == 1.0)
+                           .group_by(History.station_id.label('station_id'), func.extract('year', Obs.time).label('year'))
+                           .cte('complete'))
+
+    # This query has eligible stations with min_years or more fully complete
+    # station records
+    cte_yr_cvg = (session.query(cte_full_cvg.c.station_id,
+                                func.count(cte_full_cvg.c.station_id)
+                                )
+                        .having(func.count(cte_full_cvg.c.station_id) >= min_years)
+                        .group_by(cte_full_cvg.c.station_id)
+                        .cte('year'))
+
+    # This query uses eligible stations years from cte_full_cvg,
+    # and eligible stations from cte_yr_cvg, and pools the observations
+    # meeting both those criteria to calculate a percentile
+    query = (session.query(percentile.label('quantile'),
```

Comment:
Excellent description of what this query does. And the query itself is nicely simple and clear, made more so by the comment.